### DISCUSSION TITLE
SWT 4.x versions on Maven Central

### DISCUSSION THREAD

**Discussion by tamaracha** [eclipse-platform/eclipse.platform.swt#1370] [Q&A] [answered]
Sorry if this is a stupid question, but I am new to SWT and I am helping someone with project refactoring. The current eclipse release is [4.32](https://download.eclipse.org/eclipse/downloads/drops4/R-4.32-202406010610/), but on Maven Central, the newest version is [3.126.0](https://mvnrepository.com/artifact/org.eclipse.platform/org.eclipse.swt.cocoa.macosx.x86_64/3.126.0). Can the 4.x releases also be found on Maven Central or any maven repo?

The project author where I'm refactoring assumes that these 3.x versions on MavenCentral just get security patches from the 4.x line, but they're not the 4.x versions. He's much more experienced in using SWT than I am. My impression is that 4.32 on Eclipse download and 3.126 on MavenCentral are the same, but with different versions. The jars have aprox equal size and the manifest in the jars contain these lines:

```
Bundle-Version: 3.126.0.v20240528-0813
Eclipse-Version: 4.32
```

I'd prefer to include the 3.126 from MavenCentral in gradle, instead of downloading and including jars manually. Please tell me these are equivalent to the eclipse 4.x releases.

**Comment by merks** [comment, marked as answer]
Not all bundles is the 4.32 release are numbered 4.32.  So that is the version of that bundle. The latest version of any artifact in the org.eclipse.platform group is in general from the most recent release

**Reply by merks** [reply]
Sorry, I didn’t mean to close it. I answered on my iPhone and thought it was an issue not a discussion.

**Comment by tamaracha** [comment]
Thanks for your reply. So if the most recent version on Maven Central contains the recent version from the eclipse download page, I am fine with that answer and we can close the discussion. Then I wouldn't care about the different numbers and simply include the maven artifacts.

**Reply by merks** [reply]
Yes we publish all the artifacts for which there are new versions a day or two before the release day.
